src/main/java/com/limechain/chain/ChainService.java

```diff
@@ -50,29 +47,29 @@ protected void initialize(HostConfig hostConfig) {
             Saving the local genesis file is not suitable in this early phase of development.
             This might be removed in the future.
          */
-        isLocalChain = hostConfig.getChain() == Chain.LOCAL;
-        if (chainSpec.isPresent() && !isLocalChain) {
-            this.setGenesis((ChainSpec) chainSpec.get());
-            log.log(Level.INFO, "✅️Loaded chain spec from DB");
-            return;
-        }
+        // TODO: Think about extracting this initialization logic outside this constructor
+        Optional<Object> cachedChainSpec = repository.find(DBConstants.GENESIS_KEY);
 
-        try {
-            this.setGenesis(ChainSpec.newFromJSON(hostConfig.getGenesisPath()));
-            log.log(Level.INFO, "✅️Loaded chain spec from JSON");
+        if (cachedChainSpec.isPresent() && !isLocalChain) {
+            this.chainSpec = (ChainSpec) cachedChainSpec.get();
+            log.log(Level.INFO, "✅️Loaded chain spec from DB");
+        } else {
+            try {
+                this.chainSpec = ChainSpec.newFromJSON(hostConfig.getGenesisPath());
+                log.log(Level.INFO, "✅️Loaded chain spec from JSON");
 
-            repository.save(DBConstants.GENESIS_KEY, this.getGenesis());
-            log.log(Level.FINE, "Saved chain spec to database");
-        } catch (IOException e) {
-            throw new ChainServiceInitializationException(e);
+                repository.save(DBConstants.GENESIS_KEY, this.chainSpec);
+                log.log(Level.FINE, "Saved chain spec to database");
```

Comment:
I'm thinking that we almost always want to be avoiding anything below `INFO` level. The main idea of fine/finer/finest is to be used for logs that would spam the output and we only very occasionally would like to see being printed in order to debug, whereas messages like this one I think should be at least `INFO` level

Reply:
No probs, I've not changed that. 😄 Will now.